rust/arrow/src/compute/kernels/filter.rs

```diff
@@ -230,6 +231,86 @@ macro_rules! filter_dictionary_array {
     }};
 }
 
+macro_rules! filter_primitive_item_list_array {
+    ($context:expr, $array:expr, $item_type:ident) => {{
+        let input_array = $array.as_any().downcast_ref::<ListArray>().unwrap();
+        let values_builder = PrimitiveBuilder::<$item_type>::new($context.filtered_count);
+        let mut builder = ListBuilder::new(values_builder);
+        for i in 0..$context.filter_u64.len() {
+            // foreach u64 batch
+            let filter_batch = $context.filter_u64[i];
+            if filter_batch == 0 {
+                // if batch == 0: skip
```

Comment:
I think this comment would be improved if it explained "why" batch zero is being skipped (the code is pretty clear that batch 0 *is* being skipped). Same comment for `filter_non_primitive_item_list_array`

Reply:
the meaning of this condition is not that batch 0 is skipped, but rather that if the current batch (of 64 bits / elements from the filter array) contains only 0s (i.e. all items are filtered out) then the entire batch should be skipped as there is no items to copy to the output array (all items are filtered out); this continues until a batch is encountered where at least one item is not filtered out (and therefore the numeric representation of the batch of filter values is != 0); this skipping of entire batches of 64 items is a major factor to the performance improvements recently implemented in the filter kernel

Reply:
This makes total sense @yordan-pavlov  -- I was suggesting an improvement to the comment itself. Perhaps something like:

```suggestion
                // if batch == 0, all items are filtered out, so skip entire batch
```